snapcraft/cli/remote.py

```diff
@@ -111,43 +125,47 @@ def remote_build(
     build_id = f"snapcraft-{project.info.name}-{project_hash}"
     architectures = _determine_architectures(project, build_on)
 
+    # Calculate timeout timestamp, if specified.
+    if launchpad_timeout > 0:
+        deadline = int(time.time()) + launchpad_timeout
+    else:
+        deadline = 0
+
     lp = LaunchpadClient(
-        project=project, build_id=build_id, architectures=architectures
+        project=project,
+        build_id=build_id,
+        architectures=architectures,
+        deadline=deadline,
     )
 
     if status:
         _print_status(lp)
+        return
     elif recover:
```

Comment:
seems that if the above statement is going to be treated as a guard, this elif should start its own if block

Reply:
Fair enough, will fix